app/services/sp_return_url_resolver.rb

```diff
@@ -0,0 +1,38 @@
+class SpReturnUrlResolver
+  attr_reader :service_provider, :oidc_state, :oidc_redirect_uri
+
+  def initialize(service_provider:, oidc_state:, oidc_redirect_uri:)
```

Comment:
make these optional? then we can leave out the explicit nils in the specs:

```suggestion
  def initialize(service_provider:, oidc_state: nil, oidc_redirect_uri: nil)
```

Reply:
Added in a4f3d073c